packages/react-native-bridge/android/build.gradle

```diff
@@ -184,57 +184,52 @@ if (buildGutenbergMobileJSBundle) {
     def bundleName = 'index.android.bundle'
     def jsRootDir = '../../../..'
 
-    task bundleUpToDateCheck {
-        description("Checks if the inputs to the javascript bundle and the bundle itself are unchanged. \
+	task bundleUpToDateCheck {
+		description("Checks if the inputs to the javascript bundle and the bundle itself are unchanged. \
 If they are changed, the isBundleUpToDate flag is switched to false. That flag is used by other tasks.")
 
-        def isRelevantFile = { it.name.endsWithAny('.js', '.css') || it.name == 'package.json' }
-        def inputFiles = {
-            def jsFiles = []
-            def dirs = []
-            file(jsRootDir).eachDir { dir ->
-                if (dir.name != 'node_modules'
-                        && dir.name != 'bundle'
-                        && dir.name != 'gutenberg'
-                        && !dir.name.startsWith('symlinked')) {
-                    dirs << dir
-                }
-            }
-            file("$jsRootDir/gutenberg").eachDir { dir ->
-                if (dir.name != 'node_modules') {
-                    dirs << dir
-                }
-            }
-
-            dirs.forEach { dir ->
-                file(dir).eachFileRecurse {
-                    if (isRelevantFile(it)) {
-                        jsFiles << it
-                    }
-                }
-            }
-
-            def jsRootDirFiles = file(jsRootDir).listFiles().findAll {
-                isRelevantFile(it)
-            }
-            def gutenbergRootDirFiles = file("$jsRootDir/gutenberg").listFiles().findAll {
-                isRelevantFile(it)
-            }
-            return jsFiles + jsRootDirFiles + gutenbergRootDirFiles
-        }
-        inputs.files(inputFiles())
+		def dirs = [jsRootDir]
+		file(jsRootDir).eachDirRecurse { dir ->
+			def isRelevantDir = !['react-native-bridge/android/build/intermediates',
+								 'node_modules'].any { dir.absolutePath.contains(it) } &&
+					dir.name != 'bundle' &&
+					!dir.name.startsWith('symlinked')
+			if (isRelevantDir) {
+				dirs << dir
+			}
+		}
+
+		def isRelevantFile = { it.name.endsWithAny('.js', '.css') || it.name ==~ /package(-lock)?\.json/ }
```

Comment:
This check has been working for us so far, but I wouldn't be surprised at all if there's a filetype that we've missed. Let me know if you can think of any other "relevant" files we should be checking for changes.

Reply:
Maybe we should include ".scss", or just make it ends with "css" (without the ".")?

Reply:
👍 Good catch. I made the update. Went with adding `.scss` just for the sake of being super-explicit.